uui-components/src/widgets/Clickable.tsx

```diff
@@ -1,64 +1,68 @@
-    rawProps?: IHasRawProps<React.AnchorHTMLAttributes<HTMLAnchorElement>>['rawProps'];
+export type LinkButtonNavigationProps = {
     /** The URL that the hyperlink points to. */
     href?: never;
     /** Defines location within SPA application */
     link: Link;
 };
 
-export type ButtonRawProps = {
-    /** Any HTML attributes (native or 'data-') to put on the underlying component */
-    rawProps?: IHasRawProps<React.ButtonHTMLAttributes<HTMLButtonElement>>['rawProps'];
+export type ButtonNavigationProps = {
     /** The URL that the hyperlink points to. */
     href?: never;
     /** Defines location within SPA application */
     link?: never;
 };
 
-export type SpanRawProps = {
-    /** Any HTML attributes (native or 'data-') to put on the underlying component */
-    rawProps?: IHasRawProps<React.HTMLAttributes<HTMLSpanElement>>['rawProps'];
+export type SpanNavigationProps = {
     /** The URL that the hyperlink points to. */
     href?: never;
     /** Defines location within SPA application */
     link?: never;
 };
 
-export type AnchorRawProps = {
-    /** Any HTML attributes (native or 'data-') to put on the underlying component */
-    rawProps?: IHasRawProps<React.AnchorHTMLAttributes<HTMLAnchorElement>>['rawProps'];
+export type AnchorNavigationProps = {
     /** The URL that the hyperlink points to. */
     href: string;
     /** Defines location within SPA application */
     link: Link;
 };
 
-export type UnionRawProps = HrefRawProps | LinkButtonRawProps | ButtonRawProps | SpanRawProps | AnchorRawProps;
+export type UnionNavigationProps =
+    | HrefNavigationProps
+    | LinkButtonNavigationProps
+    | ButtonNavigationProps
+    | SpanNavigationProps
+    | AnchorNavigationProps;
+
+export type ClickableRawProps =
+    | (React.AnchorHTMLAttributes<HTMLAnchorElement>
+    | React.ButtonHTMLAttributes<HTMLButtonElement>
+    | React.HTMLAttributes<HTMLSpanElement>)
+    & Record<`data-${string}`, string>;
 
 export type ClickableComponentProps = IClickable & IAnalyticableClick & IHasTabIndex & IDisableable & IHasCX
-& ICanRedirect & UnionRawProps & {};
+& Omit<ICanRedirect, 'href' | 'link'> & UnionNavigationProps & {
+    /** Any HTML attributes (native or 'data-') to put on the underlying component */
+    rawProps?: ClickableRawProps;
+};
 
 export const Clickable = React.forwardRef<HTMLButtonElement | HTMLAnchorElement | HTMLSpanElement, PropsWithChildren<ClickableComponentProps>>((props, ref) => {
     const context = useUuiContext();
     const isAnchor = Boolean(props.href || props.link);
-    const isButton = Boolean(!isAnchor && props.onClick);
-    const isClickable = Boolean(!props.isDisabled && (isAnchor || props.onClick));
+    const isButton = Boolean(!isAnchor && ((props.rawProps as React.ButtonHTMLAttributes<HTMLButtonElement>)?.type || props.onClick));
+    const isClickable = Boolean(!props.isDisabled && (props.link || props.onClick));
```

Comment:
Why do we not include props.href in this condition?

Reply:
It uses to pass click handler to components for "link" & "onClick" props, I will rename "isClickable" to "hasClick"